2016/presentations/Python_for_Networking_Devices/text.md

```diff
@@ -0,0 +1,139 @@
+#Python for Networking Devices - Elisa Jasinska
+
+##Introduction
+
+The Internet is a series of tubes and at the end of those tubes are: networking devices! To form the Internet as we
+know it, each provider network has to be managed, maintained and interconnected. Traditional Network Engineering is
+moving more and more towards automated device and service management, a task often performed in Python due to the
+availability of many useful libraries. We will walk you though common tasks in Network Engineering and introduce a
+number of Python libraries that are helpful in accessing and managing networking equipment.
+
+##Network Device Access
+
+Traditionally networking devices are managed via their command line interface (CLI). The CLI is accessible though
+various forms of transport. Originally Telnet has been used, later on devices started to support SSH. SSH provides
+additional security mechanisms over Telnet, but its usability doesn't differ much, it's still a CLI.
+
+Configuration commands are send via the CLI, line by line, and enable or disable specific functionality on the device.
+For example, to configure a BGP neighbor on your router, your might have to add a config block similar to this one:
+```
+protocols {
+    bgp {
+        group external-peers {
+            type external;
+            neighbor 10.10.10.2 {
+                peer-as 42;
+            }
+        }
+    }
+}
+```
+
+The configurations differ per vendor, the above example would add  BGP neighbor on a Juniper device, but for a Cisco
+device, you might need to send commands like this:
+```
+router bgp 23
+neighbor 10.10.10.2 remote-as 42
+address-family ipv4 unicast
+next-hop-self
+```
+
+The drawback of entering configurations line by line is the same as with any other device (a server for instance)
+errors might occur somewhere along the way, which will result in only a part of the configuration being committed to
+the device - the change is not transactional.
+
+Nowadays Netconf is the new hype in the networking world. With underlying SSH transport it supports the same security
+feature set, but in addition it offers support  for transactions, structured data and error returns. Netconf allows to
+submit multiple changes at a time via an RPC call, encoded as XML and returns an XML object with the result in return.
```

Comment:
and sends back an XML object with the result in return

or

and returns an XML object with the result back.

(avoid "returns in return")